satpy/resample.py

```diff
@@ -509,27 +472,117 @@ def compute(self, data, fill_value=None, **kwargs):
+        # use 'mask_area' with a default of False. It wouldn't do anything.
+        return super(NativeResampler, self).resample(data,
+                                                     cache_dir=cache_dir,
+                                                     mask_area=mask_area,
+                                                     **kwargs)
+
+    @staticmethod
+    def expand_reduce(d_arr, repeats, axis):
+        if not isinstance(d_arr, da.Array):
+            d_arr = da.from_array(d_arr, chunks=CHUNK_SIZE)
+        if repeats == 1:
+            return d_arr
+        elif repeats > 1:
+            if not repeats.is_integer():
+                raise ValueError("Expand factor must be a whole number")
+            # expand
+            return da.repeat(d_arr, int(repeats), axis=axis)
+        else:
+            # reduce
+            repeats = 1. / repeats
+            if not repeats.is_integer():
+                raise ValueError("Reduce factor must be a whole number")
+            slices = tuple(slice(None, None, int(repeats))
+                           if a == axis else slice(None)
+                           for a in range(d_arr.ndim))
+            return d_arr[slices]
+
+    def compute(self, data, expand=True, **kwargs):
+        if isinstance(self.target_geo_def, (list, tuple)):
+            # find the highest/lowest area among the provided
+            test_func = max if expand else min
+            target_geo_def = test_func(self.target_geo_def,
+                                       key=lambda x: x.shape)
+        else:
+            target_geo_def = self.target_geo_def
+
+        # convert xarray backed with numpy array to dask array
+        if 'x' not in data.dims or 'y' not in data.dims:
+            if data.ndim not in [2, 3]:
+                raise ValueError("Can only handle 2D or 3D arrays without "
+                                 "dimensions.")
+            # assume rows is the second to last axis
+            y_axis = data.ndim - 2
+            x_axis = data.ndim - 1
+        else:
+            y_axis = data.dims.index('y')
+            x_axis = data.dims.index('x')
+
+        out_shape = target_geo_def.shape
+        in_shape = data.shape
+        y_repeats = out_shape[y_axis] / float(in_shape[y_axis])
+        x_repeats = out_shape[x_axis] / float(in_shape[x_axis])
+
+        d_arr = self.expand_reduce(data.data, y_repeats, y_axis)
+        d_arr = self.expand_reduce(d_arr, x_repeats, x_axis)
+
+        coords = {}
+        # Update coords if we can
+        if 'y' in data.coords or 'x' in data.coords and \
+                isinstance(target_geo_def, AreaDefinition):
+            coord_chunks = (d_arr.chunks[y_axis], d_arr.chunks[x_axis])
+            x_coord, y_coord = target_geo_def.get_proj_vectors_dask(
+                chunks=coord_chunks)
+            if 'y' in data.coords:
+                coords['y'] = y_coord
+            if 'x' in data.coords:
+                coords['x'] = x_coord
```

Comment:
this should be done in the same way as the expansion/reduction, eg decimating if the data is decimated.

Reply:
I disagree. The idea is that the output represents the same exact pixel value as the target area definition. You wouldn't want the coordinates to not match the target area in any other form of resampling.